reporter/internal/pdata/generate.go

```diff
@@ -0,0 +1,271 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pdata // import "go.opentelemetry.io/ebpf-profiler/reporter/internal/pdata"
+
+import (
+	"crypto/rand"
+	"slices"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pprofile"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/reporter/internal/samples"
+)
+
+const (
+	ExecutableCacheLifetime = 1 * time.Hour
+	FramesCacheLifetime     = 1 * time.Hour
+)
+
+// Generate generates a pdata request out of internal profiles data, to be
+// exported.
+func (p Pdata) Generate(events map[samples.TraceAndMetaKey]*samples.TraceEvents) pprofile.Profiles {
+	profiles := pprofile.NewProfiles()
+	rp := profiles.ResourceProfiles().AppendEmpty()
+
+	sp := rp.ScopeProfiles().AppendEmpty()
+
+	prof := sp.Profiles().AppendEmpty()
+	prof.SetProfileID(pprofile.ProfileID(mkProfileID()))
+	p.setProfile(events, prof)
+
+	return profiles
+}
+
+// mkProfileID creates a random profile ID.
+func mkProfileID() []byte {
+	profileID := make([]byte, 16)
+	_, err := rand.Read(profileID)
+	if err != nil {
+		return []byte("opentelemetry-ebpf-profiler")
+	}
+	return profileID
+}
+
+// setProfile sets the data an OTLP profile with all collected samples up to
+// this moment.
+func (p *Pdata) setProfile(
+	events map[samples.TraceAndMetaKey]*samples.TraceEvents,
+	profile pprofile.Profile,
+) {
```

Comment:
In the scope of message Profile the following attributes are missing:

- `default_sample_type_strindex` https://github.com/open-telemetry/opentelemetry-proto/blob/2080d87c780b5a8c65c9ae2771130fd669cef480/opentelemetry/proto/profiles/v1development/profiles.proto#L234C9-L234C37
- `period_type` https://github.com/open-telemetry/opentelemetry-proto/blob/2080d87c780b5a8c65c9ae2771130fd669cef480/opentelemetry/proto/profiles/v1development/profiles.proto#L223C13-L223C24

Reply:
I missed those in my collector upgrade. I'll add them there, thanks.

Reply:
See https://github.com/open-telemetry/opentelemetry-ebpf-profiler/issues/266